policy/pa.go

```diff
@@ -517,17 +517,25 @@ func (pa *AuthorityImpl) challengeTypesFor(identifier identifier.ACMEIdentifier)
 	var challenges []core.AcmeChallenge
 
 	// If the identifier is for a DNS wildcard name we only
-	// provide a DNS-01 challenge as a matter of CA policy.
+	// provide a DNS-based challenge as a matter of CA policy.
 	if strings.HasPrefix(identifier.Value, "*.") {
-		// We must have the DNS-01 challenge type enabled to create challenges for
-		// a wildcard identifier per LE policy.
+		// We must have a DNS-based challenge type enabled to create challenges for
+		// a wildcard identifier per CA policy.
 		if !pa.ChallengeTypeEnabled(core.ChallengeTypeDNS01) {
-			return nil, fmt.Errorf(
-				"Challenges requested for wildcard identifier but DNS-01 " +
-					"challenge type is not enabled")
+			if !pa.ChallengeTypeEnabled(core.ChallengeTypeDNSAccount01) {
```

Comment:
It seems like we can clean up this logic a little by turning it around:

 - append dns-01 if enabled
 - append dns-account-01 if enabled
 - return error if list is empty

Reply:
Done.